detox/src/server/DetoxServer.js

```diff
@@ -71,8 +77,52 @@ class DetoxServer {
     }
   }
 
-  close() {
+  async close() {
```

Comment:
The heart of the tear-down matter.
Note that the server web-socket's close() allows for giving it a callback that is essentially called when closing ends. I've tried it - it just doesn't work (called prematurely).

Reply:
Let's nevertheless try to update to `ws@^7.0.0`.
We have a too outdated `ws@^1.1.1`, and maybe they have a fix already there.
If so, we won't have to write so much workaround code.